src/ducks/sep8Send.ts

```diff
@@ -0,0 +1,165 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { Horizon } from "stellar-sdk";
+import { RootState } from "config/store";
+import { settingsSelector } from "ducks/settings";
+import { getErrorMessage } from "helpers/getErrorMessage";
+import { getErrorString } from "helpers/getErrorString";
+import { log } from "helpers/log";
+import { getToml } from "methods/getToml";
+import { submitSEP8PaymentTransaction } from "methods/sep8Send/submitSEP8PaymentTransaction";
+import {
+  ActionStatus,
+  Asset,
+  RejectMessage,
+  Sep8PaymentTransactionParams,
+  Sep8SendInitialState,
+} from "types/types.d";
+
+interface InitiateSep8SendActionResponse {
+  approvalCriteria: string;
+  approvalServer: string;
+  assetCode: string;
+  assetIssuer: string;
+  homeDomain: string;
+  isRegulated: boolean;
+}
+
+export const initiateSep8SendAction = createAsyncThunk<
+  InitiateSep8SendActionResponse,
+  Asset,
+  { rejectValue: RejectMessage; state: RootState }
+>("sep8Send/initiateSep8SendAction", async (asset, { rejectWithValue }) => {
+  const { assetCode, assetIssuer, homeDomain } = asset;
+
+  try {
+    if (!homeDomain) {
+      throw new Error("Something went wrong, home domain is not defined.");
+    }
+
+    const tomlResponse = await getToml(homeDomain);
+
+    for (let i = 0; i < tomlResponse.CURRENCIES?.length; i += 1) {
+      const currency = tomlResponse.CURRENCIES[i] as any;
+
+      if (currency.code === assetCode && currency.issuer === assetIssuer) {
+        const {
+          approval_criteria: approvalCriteria,
+          approval_server: approvalServer,
+          regulated: isRegulated,
+        } = currency;
+
+        if (!approvalCriteria) {
+          throw new Error(
+            "The anchor toml file does not contain an approval criteria.",
+          );
+        }
+
+        if (!approvalServer) {
+          throw new Error(
+            "The anchor toml file does not contain an approval server.",
+          );
+        }
+
+        // this is unlikely
+        if (!isRegulated) {
+          throw new Error(
+            'The anchor toml file does not specify this asset as "regulated".',
+          );
+        }
+
+        return {
+          approvalCriteria,
+          approvalServer,
+          assetCode,
+          assetIssuer,
+          homeDomain,
+          isRegulated,
+        };
+      }
+    }
+
+    throw new Error("Couldn't find the desired asset in the anchor toml file.");
+  } catch (error) {
+    const errorString = getErrorMessage(error);
+    log.error({
+      title: errorString,
+    });
+    return rejectWithValue({ errorString });
+  }
+});
+
+export const sep8SendPaymentAction = createAsyncThunk<
+  Horizon.TransactionResponse,
+  Sep8PaymentTransactionParams,
+  { rejectValue: RejectMessage; state: RootState }
+>(
```

Comment:
I believe I'll need to configure a custom response here (instead of `Horizon.TransactionResponse`) so we can fork the result from this duck towards multiple options.

Not 100% sure how to do that yet, but I believe it's being done for the other SEPs.